src/PackageVersions/Installer.php

```diff
@@ -124,7 +124,7 @@ private static function writeVersionClassToFile(string $versionClassSource, Comp
         $io->write('<info>ocramius/package-versions:</info>  Generating version class...');
 
         file_put_contents($installPath, $versionClassSource);
-        chmod($installPath, 0664);
+        @chmod($installPath, 0664);
```

Comment:
Should this really be silenced without an error message? It doesn't feel right.

Reply:
That's what the test would find out. IMO, since this is a CLI script modifying the data, it should probably not be silenced, but rather crash hard.

Reply:
"Hard" as in `\RuntimeException` and a `$io->write('<error>ocramius/package-versions: You ruined my life!!1!</error>')` before the exception?

Reply:
Shouldn't `file_put_contents()` just rely on what's defined by `umask`? Like:
```
↪  umask 
0002

↪ php -r 'file_put_contents("foo", "bar");'

↪ ls -l foo
-rw-rw-r-- 1 morozov morozov 3 Nov 27 12:51 foo
```

Is there a reason why the installer wants to enforce some permissions?

Reply:
http://www.doctrine-project.org/2015/08/31/security_misconfiguration_vulnerability_in_various_doctrine_projects.html

Reply:
Thanks.

Reply:
So if I set `umask 0` and run `composer install`, then the resulting files look like this:
```
↪  ls -l vendor/ocramius/package-versions/src/PackageVersions/
total 20
-rw-rw-rw- 1 morozov morozov 2582 Sep  6 08:24 FallbackVersions.php
-rw-rw-rw- 1 morozov morozov 5468 Sep  6 08:24 Installer.php
-rw-rw-r-- 1 morozov morozov 4558 Nov 27 23:06 Versions.php
```
The auto-generated file is indeed not writable by others but the rest of the files are. The attacker still can write code into other files.